index.js

```diff
@@ -1,55 +1,65 @@
-    prBody: core.getInput("pr_body"),
-    destinationBranch: core.getInput("destination_branch")
-  };
+const github = require('@actions/github');
+const core = require('@actions/core');
 
-  const base = inputs.destinationBranch;
-  const source = github.context.ref.replace(/^refs\/heads\//, "");
+const run = async () => {
+  const githubToken = core.getInput('github_token', { required: true });
+  const prTitle = core.getInput('pr_title');
+  const prBody = core.getInput('pr_body');
+  const baseBranch = core.getInput('destination_branch');
+  const sourceBranch = github.context.ref.replace(/^refs\/heads\//, '');
 
-  const octokit = github.getOctokit(inputs.githubToken);
-  core.info(`Look up a pull request with source=${source} base=${base}`);
-  const { data: pulls } = await octokit.rest.pulls.list({
+  const credentials = {
     owner: github.context.repo.owner,
     repo: github.context.repo.repo,
-    base,
-    head: `${github.context.repo.owner}:${source}`
+  };
+
+  const octokit = github.getOctokit(githubToken);
+  core.info(`Looking up a pull request with source branch="${sourceBranch || '<not found>'}" and base branch="${baseBranch || '<not specified>'}"`);
+
+  const branchHead = `${credentials.owner}:${sourceBranch}`;
+  const { data: pulls } = await octokit.rest.pulls.list({
+    ...credentials,
+    base: baseBranch,
+    head: branchHead,
   });
 
-  if (pulls.length == 0) {
-    core.info(`No such pull request: source=${source} base=${base}`);
-    return;
+  if (pulls.length === 0) {
+    throw new Error(`No pull request found for source="${source || '<not found>'}" base=${baseBranch || '<not specified>'}`);
   }
 
-  const { number } = pulls[0];
-  core.info(`Found pull request #${number}`);
+  const pullRequest = pulls.find((p) => p.state === 'open');
+  if (pullRequest == null) {
+    throw new Error(`No open pull request found for source="${source || '<not found>'}" base=${baseBranch || '<not specified>'}`);
+  }
+
+  const { number: pullNumber } = pullRequest;
+  core.info(`Found pull request #${pullNumber}`);
 
   const params = {
-    owner: github.context.repo.owner,
-    repo: github.context.repo.repo,
-    pull_number: number
+    ...credentials,
+    pull_number: pullNumber,
   };
 
-  if (inputs.prBody) {
-    core.info(`Updating with body=${inputs.prBody}`);
-    params.body = inputs.prBody;
+  if (prBody) {
+    core.info(`Updating with body="${prBody}"`);
+    params.body = prBody;
   }
-  if (inputs.prTitle) {
-    core.info(`Updating with title=#${inputs.prTitle}`);
-    params.title = inputs.prTitle;
+
+  if (prTitle) {
+    core.info(`Updating with title=#${prTitle}`);
+    params.title = prTitle;
   }
-  await octokit.rest.pulls.update(params);
-}
+
+  const url = `/repos/${credentials.owner}/${credentials.repo}/pulls/${pullNumber}`;
+
+  core.info(`Making PATCH request to "${url}" with params ${JSON.stringify(params)}`);
+  await octokit.request(`PATCH ${url}`, params);
```

Comment:
Making the request manually, because of a strange error 'Body must be a JSON' that the `octokit.pulls.update` returns

Reply:
https://docs.github.com/en/rest/reference/pulls#update-a-pull-request--code-samples

Reply:
I tested it on my other repo and it worked like a charm (and both PR title and PR body were updated as expected), while the `pulls.update` threw the above error